@xen-orchestra/proxy/src/app/mixins/backups/index.js

```diff
@@ -133,17 +132,8 @@ export default class Backups {
     app.api.addMethods({
       backup: {
         importVmBackup: [
-          defer(
-            async ($defer, { backupId, remote, srUuid, xapi: xapiOpts }) => {
-              const adapter = new RemoteAdapter(
-                await (async () => {
-                  const handler = getHandler(remote)
-                  await handler.sync()
-                  $defer.call(handler, 'forget')
-                  return handler
-                })()
-              )
-
+          defer(($defer, { backupId, remote, srUuid, xapi: xapiOpts }) =>
+            using(app.remotes.getAdapter(remote), async adapter => {
               const xapi = createXapi(xapiOpts)
               await xapi.connect()
               $defer.call(xapi, 'disconnect')
```

Comment:
Maybe it could be a resource as well :slightly_smiling_face:

Reply:
I can do it after the file restore implementation if you are OK.

Reply:
Don't worry about it, we'll do it later.